sharding/contracts/validator_manager.v.py

```diff
@@ -286,176 +284,33 @@ def get_member_of_committee(
     return self.notary_pool[sampled_index]
 
 
-# Helper function to get collation header score
-@public
-@constant
-def get_collation_header_score(shard_id: int128, collation_header_hash: bytes32) -> int128:
-    collation_score: int128 = convert(
-        uint256_mod(
-            convert(self.collation_headers[shard_id][collation_header_hash], 'uint256'),
-            # Mod 2^48, i.e., extract right most 6 bytes
-            convert(281474976710656, 'uint256')
-        ),
-        'int128'
-    )
-    return collation_score
-
-
 # Attempts to process a collation header, returns True on success, reverts on failure.
 @public
 def add_header(
+        period: int128,
```

Comment:
One sudden thought, is it feasible that we remove period from the parameters of `add_header`?
Since `period` can only be `floor(block.number / self.PERIOD_LENGTH)` if `add_header` success.

Reply:
If `period` was not specified, it is possible that headers from different period get included.

Reply:
Agree. It is more clear that pass it as parameter.